src/dsp_tools/commands/xml_validate/sparql/resource_shacl.py

```diff
@@ -79,11 +80,13 @@ def _construct_1_cardinality(onto_graph: Graph) -> Graph:
     
       ?class a owl:Class ;
           knora-api:isResourceClass true ;
+          knora-api:canBeInstantiated true ;
           rdfs:subClassOf ?restriction .
       ?restriction a owl:Restriction ;
           owl:onProperty ?propRestriction ;
           salsah-gui:guiOrder ?order ;
           owl:cardinality 1 .
+      FILTER NOT EXISTS { ?propRestriction knora-api:isLinkValueProperty true }
```

Comment:
If a property was a link property eg. `hasLinkTo` then we also get the `LinkValue` property `hasLinkToValue`, we do not want that in the cardinalities as we do not include it in the data.
Because of the in-built link properties for compound objects we need `knora-api` information.
Currently the validation does not like `knora-api` so it is only used for the construction of the SHACL but not during the validation. I am unsure if we have a problem in `knora-api` or if it simply does not conform to TopBraid's expectations...

Reply:
I guess we should take some time then investigating the knora-api ontology.

Reply:
Yes, my plans are:
- Ask stackoverflow how inheritance with closed shapes should be approached, because I understand that it is actually contradictory. Perhaps we can hinder SHACL inheritance (shapes are also inherited, just as property cardinalities are)
- There is another weird behaviour regarding property inheritance, I am not sure if TopBraid is doing something weird or we are, I need to check further but I think in our data it is an edge case.
- Investiage `knora-api` because I don't think it is possible not to include it once we are implementing the built-in dsp classes. But that comes last so in the mean time I think it is ok like this.

We can discuss it further on Monday.